### PR TITLE
feat: implement max property and overlay

### LINKED ISSUE #7: vaadin-avatar-group: handle too many avatars to display
Add `max` property for the maximum number of avatars to display. The overflowing N avatars are grouped into one avatar with text "+N":

The vcf component has the same API: https://vcf-avatar-group.netlify.app/

We might want to consider providing the option to dynamically overflow avatars based on the avatar-group's width, if there appears to be demand for this kind of functionality.

### LINKED ISSUE #8: Theming
For avatar and avatar-group
- Lumo
- Material
- should use user colors defined by the theme
- themable parts?
  - not for now unless we come up with something important

### PR DESCRIPTION
**Pull request by web-padawan**
Fixes #7 
Fixes #8

### CODE REVIEW

**Review by jouni — CHANGES_REQUESTED** [review]
Behavior looks great.

Lumo:
- The avatars in the overflow menu should use a smaller size variant and some margins, so that the item height is kept at `--lumo-size-m` (36px)
- The visual distance of the avatar and the left edge of the overlay should be `--lumo-space-s` (8px) (the overlay has a 4px padding already).
- The same applies for the visual distance of the avatar and the user name (in the overlay)

**Comment by web-padawan** [comment]
Rebased and updated visual tests to pick up changes from master. @jouni PTAL

File: src/vaadin-avatar-group.html [hunk 1]
```diff
@@ -33,14 +36,48 @@
...
+      <template is="dom-repeat" items="[[__computeItems(items.*, max)]]">
+        <vaadin-avatar
+          name="[[item.name]]"
+          abbr="[[item.abbr]]"
+          img="[[item.img]]"
+          part="avatar"
+          theme$="[[theme]]"
+        ></vaadin-avatar>
       </template>
+      <vaadin-avatar
```
Comment (yuriy-fix) [inline src/vaadin-avatar-group.html:48]: Probably can be fixed separately, but currently it has problems with calculating `margins` as it's calculated based on number of `items` and this one is not a part of those.

File: src/vaadin-avatar-group.html [hunk 2]
```diff
@@ -78,20 +115,140 @@
...
+            /**
+             * The maximum number of avatars to display.
+             * By default, all the avatars are displayed.
+             * When max is set, the overflowing avatars are grouped into one avatar.
+             */
+            max: {
+              type: Number
+            },
+
+            _maxReached: {
```
Comment (yuriy-fix) [inline src/vaadin-avatar-group.html]: Nit: should it be private instead? 🤔

File: src/vaadin-avatar-group.html [hunk 3]
```diff
@@ -78,20 +115,140 @@
...
+            if (/^(Enter|SpaceBar|\s)$/.test(e.key)) {
+              e.preventDefault();
+              this._opened = true;
+            }
+          }
+        }
+
+        _onListKeyDown(event) {
+          if (event.key === 'Escape' || event.key === 'Esc') {
+            this._opened = false;
```
Comment (yuriy-fix) [inline src/vaadin-avatar-group.html:174]: I guess overlay should be closed whenever we are leaving tabs or we will get into the case where after tabbing out of the list-box you cannot properly close it anymore (until moving focus back):
close-overlay

File: theme/lumo/vaadin-avatar-styles.html [hunk 4]
```diff
@@ -14,12 +14,18 @@
...
+        outline: none;
         cursor: default;
         user-select: none;
         -webkit-font-smoothing: antialiased;
         -moz-osx-font-smoothing: grayscale;
       }
 
+      /* TODO: does not work with <img> */
+      :host([focus-ring]) {
+        box-shadow: inset 0 0 0 2px var(--lumo-primary-color-50pct);
```
Comment (yuriy-fix) [inline theme/lumo/vaadin-avatar-styles.html:26]: Nit: we can create an issue right away. Just recorded 😅 
focus-ring

**Review by jouni — APPROVED** [review]
Visually, looks good now.

I would still add a little bit of additional space after the name in the overlay (similarly [as we do list-box](https://github.com/vaadin/vaadin-list-box/blob/master/theme/lumo/vaadin-list-box-styles.html#L34)), so that the text does not run into the edge of the overlay.

File: src/vaadin-avatar-group.html (hunk 1 above)
Comment (yuriy-fix) [inline src/vaadin-avatar-group.html:48]: nit: should it have different title? not `anonymous`? 🤔

**Comment by web-padawan** [comment]
> I would still add a little bit of additional space after the name in the overlay (similarly as we do list-box), so that the text does not run into the edge of the overlay.

Agree. Updated to use `padding-right: var(--lumo-space-m)` so there will be some space.

File: src/vaadin-avatar-group.html (hunk 1 above)
Comment (web-padawan) [reply src/vaadin-avatar-group.html:48]: Yes, let's fix it separately. As discussed in Slack, the whole approach might need some changes.

File: theme/lumo/vaadin-avatar-styles.html (hunk 4 above)
Comment (web-padawan) [reply theme/lumo/vaadin-avatar-styles.html:26]: Thanks, created https://github.com/vaadin/vaadin-avatar/issues/25

File: src/vaadin-avatar-group.html (hunk 1 above)
Comment (web-padawan) [reply src/vaadin-avatar-group.html:48]: Thanks, fixed.

File: src/vaadin-avatar-group.html (hunk 2 above)
Comment (web-padawan) [reply src/vaadin-avatar-group.html]: Changed.

File: src/vaadin-avatar-group.html (hunk 3 above)
Comment (web-padawan) [reply src/vaadin-avatar-group.html:174]: Good catch. Fixed, added missing tests.

**Review by yuriy-fix — APPROVED** [review]

**Pull request merged**